### PR TITLE
Added the changes that we worked on Sat, also added the skip. Moved code from from Main to Import Dataset.

### PR DESCRIPTION
**Pull request by deaspo**
Added the changes that we worked on Sat, also added the skip. Moved code from from Main to Import Dataset.

### CODE REVIEW

File: instat/dlgImportDataset.vb [hunk 1]
```diff
@@ -160,9 +165,11 @@ Public Class dlgImportDataset
         Dim bToBeAssigned As Boolean
         bToBeAssigned = ucrBase.clsRsyntax.clsBaseFunction.bToBeAssigned
         ucrBase.clsRsyntax.clsBaseFunction.bToBeAssigned = False
-        ucrBase.clsRsyntax.AddParameter("nrows", 10)
+        ucrBase.clsRsyntax.AddParameter("nrows", numLines.Value)
+        ucrBase.clsRsyntax.AddParameter("skip", numSteps.Value)
```
Comment (dannyparsons) [inline instat/dlgImportDataset.vb]: "skip" is the correct R parameter but is not the same as the way you have used numSteps. Have a look at the R help for read.csv to understand skip or happy to discuss.

File: instat/dlgImportDataset.vb [hunk 2]
```diff
@@ -160,9 +165,11 @@ Public Class dlgImportDataset
...
         dfTemp = frmMain.clsRLink.GetData(ucrBase.clsRsyntax.GetScript())
         ucrBase.clsRsyntax.RemoveParameter("nrows")
+        ucrBase.clsRsyntax.RemoveParameter("skip")
```
Comment (dannyparsons) [inline instat/dlgImportDataset.vb]: skip should never be removed from the RSyntax. It's always needed.

File: instat/dlgImportDataset.vb [hunk 3]
```diff
@@ -186,5 +193,61 @@ Public Class dlgImportDataset
...
+                    'TODO where should this go?
+                    If Not frmMain.clsRLink.bInstatObjectExists Then
+                        frmMain.clsRLink.CreateNewInstatObject()
+                    End If
+                    SetName(pair.Key)
+                    SetFilePath(pair.Value)
+                    strFilePath = pair.Value
+            End Select
+        End If
+    End Sub
```
Comment (dannyparsons) [inline instat/dlgImportDataset.vb]: Great that this has been moved here from frmMain. We can probably tidy this up a lot now.

There should also be a text box (possibly un-editable) for the file path. Do we have that yet?

File: instat/dlgImportDataset.vb [hunk 4]
```diff
@@ -61,7 +66,7 @@ Public Class dlgImportDataset
         Dim sReader As New StreamReader(strFilePath)
         ucrBase.clsRsyntax.AddParameter("file", Chr(34) & strFilePath & Chr(34))
         txtInputFile.Text = ""
-        For i = 1 To 10 'TODO add skip and get 10 from options
+        For i = 1 To numLines.Value Step numSteps.Value + 1
```
Comment (dannyparsons) [inline instat/dlgImportDataset.vb]: numLines should not be a value on the dialog that the user can change. The number of lines of the file to read for the preview should be an option in the Options dialog (with the default being 10). So you can delete numLines from this dialog.

Also, I don't understand the use of Step and numStep. Why have you added this option? We should instead have a numeric up/down for "skip" on the dialog and this should be added as a parameter to the function. 
Then the loop goes from 1 To 10 + (skip value).

**Comment by dannyparsons** [comment]
Still some things needed here - see the comments. If anything's not clear let me know so we can discuss. Thanks

File: instat/dlgImportDataset.Designer.vb [hunk 5]
```diff
@@ -47,12 +47,19 @@ Partial Class dlgImportDataset
         Me.chlStringsAsFactors = New System.Windows.Forms.CheckBox()
         Me.dataFrame = New unvell.ReoGrid.ReoGridControl()
         Me.ucrBase = New instat.ucrButtons()
+        Me.lblSteps = New System.Windows.Forms.Label()
+        Me.lblLines = New System.Windows.Forms.Label()
+        Me.numLines = New System.Windows.Forms.NumericUpDown()
+        Me.numSteps = New System.Windows.Forms.NumericUpDown()
+        Me.cmdOpenDataSet = New System.Windows.Forms.Button()
```
Comment (dannyparsons) [inline instat/dlgImportDataset.Designer.vb]: I think the convention for numeircupdown is "nud", but maybe check this with Mary first.

File: instat/dlgImportDataset.vb (hunk 4 above)
Comment (deaspo) [inline instat/dlgImportDataset.vb]: I'd like to discuss this with you, when you're available on skype, let me know.

**Pull request closed without merging**